deepspeed/runtime/engine.py

```diff
@@ -2804,10 +2814,17 @@ def _load_zero_checkpoint(self, load_dir, tag, load_optimizer_states=True):
             if zero_sd_list is None:
                 return False
 
-        self.optimizer.load_state_dict(state_dict_list=zero_sd_list,
-                                       load_optimizer_states=load_optimizer_states,
-                                       load_from_fp32_weights=self.zero_load_from_fp32_weights(),
-                                       checkpoint_folder=checkpoint_folder)
+        if load_serial is not None:
```

Comment:
If this `if` condition needed since we are passing `load_serial` along?

Reply:
Yep, it could be simplified. I would send another patch to fix it, since I found original patch already merged?